app/models/enhancer.rb

```diff
@@ -0,0 +1,17 @@
+class Enhancer
+  attr_accessor :enhanced_query
+
+  # accepts all params as each enhancer may require different data
+  def initialize(params)
+    @enhanced_query = {}
+    @enhanced_query[:raw_query] = params
```

Comment:
Some part of me is nervous about blindly trusting everything that a user might pass in as a querystring - would it be preferrable to deal only with parameters which we know and expect might exist here?

Reply:
This is just a convenient way to leave the unmodified version around. We will still just pull out what we expect to be used in various analyzers/builders. Does that help or am I missing something about what you are concerned with?

Reply:
I'm worried about future-us doing something with untrusted user input, and whether it being an attribute of an `enhanced_query` might lull us into thinking that it is trustworthy. The "raw" prefix in the name should help this, I think - and I may be overthinking this in general.

If I understand the application flow correctly, the intent is that the enhanced_query hash is what's going to be sent to the QueryBuilder, and then to the TimdexWrapper? Somewhere along the line, I think I'd prefer for there to be some sort of whitelist/filter for fields - particularly if we're going to eventually pass user input down into the Timdex UI or other deeper systems, especially of their APIs aren't read-only.

(Although I'm also realizing that Timdex is a read-only system, so I don't think it matters what we throw at it via a GET request?)

I may be overthinking this - if all of this doesn't resonate with you, then I'm happy to approve the PR as-is.

Reply:
Not trusting user input as a default stance is good security practice. However, copying the params to another variable and including it in a hash changes nothing to our security practices. If renaming to make it clear that `raw_query` is indeed just the request params, I'd be happy to do that. Would `raw_params` or `request_params` feel less likely to be mistaken for safe data to you?

For additional context, I felt we had two options:

- pass the params around separately from the enhanced_query as things downstream will need different bits and we don't know what yet
- store the params inside the enhanced_query so we only pass one thing around and downstream methods that need to pull a param can do so but things that don't won't need to even know params exist

I'm also not opposed to moving towards pulling out everything we need like we do `q` (essentially creating the safelist of params but not actually without explicit action making sure they are actually safe) but I think I'd like to sit on that for a bit as that seems more likely to be a concern of the QueryBuilder and not the Enhancers (although the Enhancers might end up caring about additional params... but I'm not sure what yet?)

Finally, nothing I am doing in this PR uses `raw_query`. I don't mind removing it entirely until we get a better feel for whether it makes sense. It was added as "I suspect this will be useful later" but it would be super easy to add back in later if that's the case.

Reply:
I'd prefer the name `raw_query` over `request_params`, which you're already using - and I'm not coming up with a better alternative that doesn't introduce its own problems.

While it might be preferable to introduce this only when we have a way of using it, we're at the stage of the project where things aren't expected to be super clean and tidy? If you think there's going to be a general cleanup in the future to optimize things, then I'd say we can merge this now.

In that vein, I'll approve the PR as-is.

Reply:
Thanks for this conversation. For now I'm going to remove the unused affordance and when/if we need it we'll have more context for whether it introduces concerns and what to do about them.